lua/compe_tabnine/init.lua

```diff
@@ -181,6 +181,7 @@ Source._on_stdout = function(_, data, _)
 							local item = {
 								word = result.new_prefix;
 								user_data = result;
+								filter_text = old_prefix;
```

Comment:
Im really struggling with this. Most of this stuff is not documented. What exactly does `filter_text` do?

Reply:
`filter_text`, when set, will be text that gets compared against the prefix in compe. It is set to old_prefix so that these items don't get filtered out *at all*. For example, in when you enter `passwrod`, you could get a completion suggestion for `password` but it will get filtered out by compe's filtering algorithm. By setting `filter_text` this way it basically makes all the items go through without being filtered. I do find these suggestions from tabnine quite useful so i decided to get all of the suggestions through. 

In vim this behaviour would have been implemented using the `equal` complete-item attribute but compe ignores it and has its own mechanism to (fuzzy) filter, this is why this hacky-looking text attribute is used.

Reply:
I wonder why compe is not documented at all. 
This makes sense looking at compe/matcher.lua